### DISCUSSION TITLE
Add switch to Home Assistant (e.g. for electric heater)

### DISCUSSION THREAD

**Discussion by EarlSneedSinclair** [philippoo66/optolink-splitter#64] [Q&A] [answered]
Hi,

I would like to add a switch for the CU401B-S's electric heater to Home Assistant.

Following adresses need to be written:
- STATE: ```r;0x7902;1;1;False```
- ON: ```w;0x7902;1;1```
- OFF: ```w;0x7902;1;0```

Added to settings_ini.py: ```("eheizung_betrieb", 0x7902, 1, 1, False), # WPR_eHeizung_Heizbetrieb_mit_Elektro```

Added to discovery json file: 

```json
{
	"domain": "switch",
	"name": "Durchlauferhitzer",
	"state_topic": "eheizung_betrieb",
	"mode_command_topic": "cmnd",
	"payload_on": "w;0x7902;1;1",
	"payload_off": "w;0x7902;1;0",
	"state_on": "1",
	"state_off": "0"
},
```

I believe some modifications to HA_discovery.py are required, but I am struggling to get it right. Could anyone provide guidance or suggestions on how to proceed?

Thank you in advance!

**Comment by FransOv** [comment, marked as answer]
I think the only thing that is not correct is that mode_command_topic should be command_topic. See also: https://www.home-assistant.io/integrations/switch.mqtt/

**Comment by EarlSneedSinclair** [comment]
```json
{
	"domain": "switch",
	"name": "Durchlauferhitzer",
	"state_topic": "eheizung_betrieb",
	"command_topic": "cmnd",
	"payload_on": "w;0x7902;1;1",
	"payload_off": "w;0x7902;1;0",
	"state_on": "1",
	"state_off": "0"
},
```

Yes, that was the issue. The snippet above is working. Many thanks for your quick reply, and a Happy New Year to you.
